### PR TITLE
Tabs: make inner components customizable.

### PR DESCRIPTION
**Pull request by alansouzati**
#### What does this PR do?

Changes tab and tabs to allow customizable inner components

#### Where should the reviewer start?
Tabs.js and Tab.js

#### What testing has been done on this PR?
manual
#### How should this be manually tested?
open storybook examples for tabs. especially custom theme one.
#### What are the relevant issues?
https://github.com/grommet/grommet/issues/2336
#### Do the grommet docs need to be updated?
yes
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards

### CODE REVIEW

File: src/js/components/Tab/Tab.js [hunk 1]
```diff
@@ -84,13 +95,26 @@ class Tab extends Component {
...
+            {header}
+          </StyledTab>
+        ) : (
+          <StyledTab
+            as={Box}
+            background={theme.tab.background}
+            theme={theme}
+            pad={{ bottom: 'xsmall' }}
+            margin={{ vertical: 'xxsmall', horizontal: 'small' }}
+            border={{ side: 'bottom', size: 'small', color: borderColor }}
```
Comment (ericsoderberghp) [inline src/js/components/Tab/Tab.js]: Would one nice if this came from the theme rather than here.

File: src/js/components/Tab/doc.js [hunk 2]
```diff
@@ -9,6 +9,9 @@ export const doc = (Tab) => {
     );
 
   DocumentedTab.propTypes = {
+    header: PropTypes.node.description(
+      'If specified, the entire panel header will be managed by the caller.'
```
Comment (ericsoderberghp) [inline src/js/components/Tab/doc.js]: I'm not sure what 'panel' refers to in this context. I'm also wondering if `header` is the right term. I'm wondering if we should call it `label` instead?

Comment (alansouzati) [reply src/js/components/Tab/doc.js]: I'm following the pattern in Accordion. we call `header` there.

Comment (alansouzati) [reply src/js/components/Tab/doc.js]: I agree with the wording let me change it

File: src/js/components/Tab/Tab.js (hunk 1 above)
Comment (alansouzati) [reply src/js/components/Tab/Tab.js]: checking...

File: src/js/components/Tab/doc.js (hunk 2 above)
Comment (alansouzati) [reply src/js/components/Tab/doc.js]: actually it makes sense your feedback. in accordion, it is actually a header. here it is not.

**Pull request merged**